service/src/main/kotlin/fi/espoo/evaka/units/UnitsView.kt

```diff
@@ -112,7 +116,8 @@ class UnitsView(private val accessControl: AccessControl) {
                         placements.flatMap { placement ->
                             placement.groupPlacements.mapNotNull { groupPlacement -> groupPlacement.id }
                         }
-                    )
+                    ),
+                    unitChildrenCapacityFactors = if (accessControl.hasPermissionFor(user, Action.Unit.READ_CHILD_CAPACITY_FACTORS, unitId)) capacities else listOf()
```

Comment:
It would make more sense to do the access control check before querying the data to minimize unnecessary work.